Esri/arcgis-maps-sdk-swift-samples
# [New] Add dynamic entity layer

## Description

This PR implements `Add dynamic entity layer` in `Maps` category.
URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/mhd/AddDynamicEntityLayer/Shared/Samples/Add%20dynamic%20entity%20layer/README.md

## Linked Issue(s)

- `swift/issues/3734 `

## How To Test

- Tap "Disconnect/Connect" button and make sure connection status and observations are updated appropriately. When disconnected, observations will not be updated on the map.
- Open settings
- Toggle "Track lines" and "Previous observations" and ensure map updates appropriately
- Change "Observations per track" and ensure map updates to match
- Tap "Purge all observations" and ensure all observations are removed.

## To Discuss

Should the Model and/or Settings view go in separate files?
Doe the UI meet expectations and match existing samples?
